Sources/Fluent/Entity/Entity.swift

```diff
@@ -19,6 +19,15 @@ public protocol Entity: Preparation, NodeConvertible {
         like pivots.
     */
     static var name: String { get }
+    
+    /**
+        The name of the column that corresponds
+        to this entity's key.
+     
+        Return nil to use the database's global
+        idKey.
+     */
+    static var idKey: String? { get }
```

Comment:
I'm wondering if this should be `String` and just return the `database.driver.idKey` by default so that we can deal with this everywhere internally w/o user additions.